UnitTests/Tests.cs

```diff
@@ -1043,7 +1029,7 @@ public void SafeReadByteDoesNotDependOnStreamPosition()
             var buffer = this.GetRandomBuffer(stream.Capacity * 2);
             stream.Write(buffer, 0, buffer.Length);
 
-            for (var i = 0; i < stream.Length; i++)
+            for (long i = 0; i < stream.Length; i++)
             {
```

Comment:
Why the change? `var` assumes the correct type.

You can keep `var` if you use the correct literal: `0L`.